### PR TITLE
Update librsvg to version 2.59.1

### PR DESCRIPTION
**Pull request by danyeaw**
- librsvg updated the build system to meson
- Update Rust to 1.82.0
- Patch librsvg to work around issue with renaming archive file from an upstream MR

### CODE REVIEW

**Comment by pbor** [comment]
It is not clear to me if it is the same issue or an additional one, but there is also a compatibility issue with meson 1.5 https://gitlab.gnome.org/GNOME/librsvg/-/issues/1099

**Comment by danyeaw** [comment]
Upstream Rust issue: https://github.com/rust-lang/rust/issues/129020

**Comment by pbor** [comment]
If that is accurate, the fix should be in Rust 1.82.0, which is scheduled on Oct 17

**Comment by fanc999-1** [comment]
> It is not clear to me if it is the same issue or an additional one, but there is also a compatibility issue with meson 1.5 https://gitlab.gnome.org/GNOME/librsvg/-/issues/1099

No, it's an additional one that actually hits HarfBuzz, not librsvg. I checked with 1.5.1, and things worked there.

**Comment by pbor** [comment]
> > It is not clear to me if it is the same issue or an additional one, but there is also a compatibility issue with meson 1.5 https://gitlab.gnome.org/GNOME/librsvg/-/issues/1099
> 
> No, it's an additional one that actually hits HarfBuzz, not librsvg. I checked with 1.5.1, and things worked there.

Ok, but if I read it correctly that only affects meson subprojects so it should not affect us

File: gvsbuild/projects/librsvg.py [hunk 1]
```diff
@@ -36,29 +36,27 @@ def __init__(self):
...
-        self.push_location(b_dir)
-        self.exec_vs(cmd)
-        self.pop_location()
+        self.add_param(f"-Dintrospection={enable_gi}")
+        self.add_param("-Ddocs=disabled")
+        self.add_param("-Dtests=false")
+        self.add_param("-Dvala=disabled")
 
+    def build(self):
+        # Install cargo-c version compatible with Cargo 0.77.x
```
Comment (pbor) [inline gvsbuild/projects/librsvg.py:57]: is this still needed if now we can use the latest Rust compiler?

Comment (danyeaw) [reply gvsbuild/projects/librsvg.py]: Hi @pbor, good catch. We still need to install cargo-c, but I unpinned this older version.

**Review by pbor — APPROVED** [review]

**Pull request merged**